python/tk_framework_desktopserver/shotgun/api_v2.py

```diff
@@ -132,6 +133,32 @@ def execute_action(self, data):
 
         :param dict data: The payload from the client.
         """
+        # When the v2 protocol is being used, the Shotgun web app makes use of
+        # Javascript Promises to handle the reply from the server, whether there
+        # is success or failure. When getting actions, we want to make sure that
+        # if there was some kind of unhandled exception that there is a proper
+        # reply to the client so that the Promise can be kept or broken, as is
+        # appropriate.
+        try:
+            self._execute_action(data)
+        except Exception:
+            import traceback
+            self.host.reply(
+                dict(
+                    err="Failed to execute action: %s" % traceback.format_exc(),
```

Comment:
I'd like to get our security experts opinion at some point about returning Python error messages back to the browser. I wonder if this isn't some sort of security issue because you are telling an attacker how he broke your server.

Reply:
Good point. I'll bring it up with Rob next time he and I have our 1:1.